pkg/api/steve/supportconfigs/generator.go

```diff
@@ -24,34 +25,114 @@ import (
 
 const (
 	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
-	Endpoint            = "/v1/generateSUSERancherSupportConfig"
+	Endpoint = "/v1/generateSUSERancherSupportConfig"
+	// NOTE: the name of the configmap will be the same for both Bring-Your-Own (BOY) and
+	// Pay-As-You-Go (PAYG) offerings. However, they will be in different namespaces to avoid
+	// clashing.
 	cspAdapterConfigmap = "csp-config"
 	tarContentType      = "application/x-tar"
 	logPrefix           = "support-config-generator"
 )
 
 var errNotFound = errors.New("not implemented")
 
-// Handler implements http.Handler - and serves supportconfigs (tar file which contains support relevant information)
-type Handler struct {
+type cspAdapterInterface interface {
+	GetRelease(chartNamespace string, chartName string) (*release.Release, error)
+}
+
+type cspConfigInterface interface {
+	authorize(cspNamespace string, r *http.Request) (bool, error)
+	getCSPConfig(cspNamespace string) (map[string]interface{}, error)
+}
+
+type CSPConfig struct {
 	ConfigMaps           v1.ConfigMapInterface
 	SubjectAccessReviews authv1.SubjectAccessReviewInterface
-	adapterUtil          *cspadapter.ChartUtil
+}
+
+// authorize checks to see if the user can get the csp adapter configmap. Returns a bool (if the user is authorized)
+// and optionally an error
+func (c *CSPConfig) authorize(cspNamespace string, r *http.Request) (bool, error) {
+	userInfo, ok := request.UserFrom(r.Context())
+	if !ok {
+		return false, fmt.Errorf("unable to extract user info from context")
+	}
+	response, err := c.SubjectAccessReviews.Create(r.Context(), &authzv1.SubjectAccessReview{
+		Spec: authzv1.SubjectAccessReviewSpec{
+			ResourceAttributes: &authzv1.ResourceAttributes{
+				Resource:  "configmap",
+				Verb:      "get",
+				Name:      cspAdapterConfigmap,
+				Namespace: cspNamespace,
+			},
+			User:   userInfo.GetName(),
+			Groups: userInfo.GetGroups(),
+			Extra:  convertExtra(userInfo.GetExtra()),
+			UID:    userInfo.GetUID(),
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to create sar %s", err)
+	}
+	if !response.Status.Allowed {
+		return false, nil
+	}
+	return true, nil
+}
+
+// getCSPConfig gets the configmap produced by the csp-adapter returns an error if not able to produce the map. Will return
+// an errNotFound if the map is not found at all
+func (c *CSPConfig) getCSPConfig(cspNamespace string) (map[string]interface{}, error) {
```

Comment:
Similar comment here as with the authorize function, I don't think this needs to be moved to a separate struct.